api/db/seeds/data/team-certification/cases/simple-pro-certification.js

```diff
@@ -1,206 +1,166 @@
-    accessCode: 'AZERTY',
-  });
-
-  const publishedScoCandidateId = await enrolmentUseCases.addCandidateToSession({
-    sessionId: PUBLISHED_PRO_SESSION,
-    candidate: new Candidate(candidate), // Warning: usecase modifies the entry model...
-    normalizeStringFnc: normalize,
-  });
-
-  await publishSessionWithValidatedCertification({
-    databaseBuilder,
-    sessionId: PUBLISHED_PRO_SESSION,
-    candidateId: publishedScoCandidateId,
-    pixScoreTarget: 250,
-  });
+export class ProSeed {
+  constructor({ databaseBuilder }) {
+    this.databaseBuilder = databaseBuilder;
+  }
+
+  async create() {
+    const organizationMember = await this.#addOrganization();
+    const { certificationCenter, certificationCenterMember } = await this.#addCertifCenter({ organizationMember });
+    const certifiableUser = await this.#addCertifiableUser();
+
+    /**
+     * Session with candidat ready to start his certification
+     */
+    const sessionReadyToStart = await this.#addReadyToStartSession({ certificationCenterMember, certificationCenter });
+    await this.#addCandidateToSession({ pixAppUser: certifiableUser, session: sessionReadyToStart });
+
+    /**
+     * Session with a published certification
+     */
+    const sessionToPublish = await this.#addSessionToPublish({ certificationCenterMember, certificationCenter });
+    const candidateToPublish = await this.#addCandidateToSession({
+      pixAppUser: certifiableUser,
+      session: sessionToPublish,
+    });
+
+    await publishSessionWithValidatedCertification({
+      databaseBuilder: this.databaseBuilder,
+      sessionId: PUBLISHED_PRO_SESSION,
+      candidateId: candidateToPublish.id,
+      pixScoreTarget: 250,
+    });
+  }
+
+  async #addOrganization() {
+    const commonOrgaService = await CommonPixCertifOrganization.getInstance({ databaseBuilder: this.databaseBuilder });
+    return commonOrgaService.organizationMember;
+  }
+
+  async #addCertifCenter({ organizationMember }) {
+    const certificationCenter = await organizationalEntitiesUsecases.createCertificationCenter({
+      certificationCenter: new CertificationCenter({
+        id: PRO_CERTIFICATION_CENTER_ID,
+        name: 'PRO Certification Center',
+        type: certificationCenterTypes.PRO,
+        externalId: PRO_CERTIFICATION_CENTER_EXTERNAL_ID,
+        createdAt: new Date('2022-01-30'),
+        habilitations: [],
+        isV3Pilot: true,
+      }),
+      complementaryCertificationIds: [],
+    });
+
+    const certificationCenterMember = await teamUsecases.createCertificationCenterMembershipByEmail({
+      certificationCenterId: certificationCenter.id,
+      email: organizationMember.email,
+    });
+
+    return { certificationCenter, certificationCenterMember };
+  }
+
+  async #addCertifiableUser() {
+    const certifiableUserService = await CommonCertifiableUser.getInstance({ databaseBuilder: this.databaseBuilder });
+    return certifiableUserService.certifiableUser;
```

Comment:
```suggestion
    const { certifiableUser } = await CommonCertifiableUser.getInstance({ databaseBuilder: this.databaseBuilder });
    return certifiableUser;
```
(une proposition qui n'est qu'un détail)

Reply:
done